gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/FlowSpec.java

```diff
@@ -125,6 +125,18 @@ public static FlowSpec.Builder builder(URI catalogURI, Properties flowProps) {
       throw new RuntimeException("Unable to create a FlowSpec URI: " + e, e);
     }
   }
+
+  /**
+   * Add new property at the specified path to the configAsProperties objects.
+   * Note: this does NOT update the Config so any property added through this function must be retrieved through the
+   * ConfigAsProperties field
```

Comment:
may be error prone... why not instead create a new `Config` and update the reference to point to that one?  given immutability, k-v pairs in common (the vast majority) should be readily shared, for minimal performance penalty, so long as you create the new config using `Config::withFallback` 

do be sure to synchronize an update of `configAsProperties` at that same time.